### PR TITLE
Add an edit button to edit an external device configuration (#5950)

### LINKED ISSUE #5950: Add an edit button to edit an external device configuration
Add an edit button for each line that open a popup for edition (similar to other admin screen)

image

### PR DESCRIPTION
**Pull request by vlo-rte**
The issue mentioned by sonarcloud is not related to the work done for this issue.

- In release note :
  -  In chapter : Features
  -  Text : #5950 : Add an edit button to edit an external device configuration

### CODE REVIEW

File: src/test/api/karate/externaldevices/createDeviceConfig.feature [hunk 1]
```diff
@@ -31,7 +31,7 @@ Feature: Device Configuration Management (Create)
     And header Authorization = 'Bearer ' + authToken
     And request configuration
     When method post
-    Then status 400
+    Then status 201
```
Comment (quinarygio) [inline src/test/api/karate/externaldevices/createDeviceConfig.feature:34]: The Scenario description should be "Update an existing configuration" ?
And as it is an update I would expect a 200 status code

File: services/external-devices/src/main/java/org/opfab/externaldevices/controllers/ConfigurationsController.java [hunk 2]
```diff
@@ -71,7 +71,7 @@ public Void createDeviceConfiguration(HttpServletRequest request, HttpServletRes
             @Valid @RequestBody DeviceConfiguration deviceConfiguration) {
 
         String id = deviceConfiguration.id;
-        configService.insertDeviceConfiguration(deviceConfiguration);
+        configService.saveDeviceConfiguration(deviceConfiguration);
```
Comment (quinarygio) [inline services/external-devices/src/main/java/org/opfab/externaldevices/controllers/ConfigurationsController.java:74]: When it is an update it should return a 200 status code

**Review by quinarygio — CHANGES_REQUESTED** [review]

**Comment by quinarygio** [comment]
When editing a "disabled" device, the modal has always the checkbox "is enabled" checked

**Comment by freddidierRTE** [comment]
@quinarygio  :  You've made a valid point about the return code.  However, I suggest we maintain the current code for these reasons:

- This API is not intended for third-party use.
- The feature in question is minor.
- It doesn't affect reliability or security.
- Addressing this would require time, and there are other higher-priority tasks to focus on.

**Comment by vlo-rte** [comment]
> When editing a "disabled" device, the modal has always the checkbox "is enabled" checked

Problem solved. Please check again

File: src/test/api/karate/externaldevices/createDeviceConfig.feature (hunk 1 above)
Comment (vlo-rte) [reply src/test/api/karate/externaldevices/createDeviceConfig.feature:34]: You're right, I've renamed the scenario description.

**Review by quinarygio — APPROVED** [review]

**Pull request merged**